src/StreamTrait.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/**
+ * @see       https://github.com/laminas/laminas-diactoros for the canonical source repository
+ * @copyright https://github.com/laminas/laminas-diactoros/blob/master/COPYRIGHT.md
+ * @license   https://github.com/laminas/laminas-diactoros/blob/master/LICENSE.md New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Laminas\Diactoros;
+
+trait StreamTrait
+{
+    /**
+     * A list of types that are allowed to instantiate a Stream
+     */
+    private $allowedTypes = ['gd', 'stream'];
```

Comment:
This should rather be a private constant (which works great with the centralisation of the validation on the stream class).

Let's also remember that properties on traits are replicated in every instance, which might be a not so great usage of memory.

Reply:
@lcobucci, I took your advice and rolled the functionality from StreamTrait into Stream. Thanks for your guidance.